controllers/chat/findOrCreateChannelBySignedSender.js

```diff
@@ -0,0 +1,48 @@
+const {StreamChat} = require('stream-chat');
+const {CHANNEL_TYPE_STRING, CHANNEL_TYPE} = require('../../helpers/constants');
+const ErrorResponse = require('../../utils/response/ErrorResponse');
+
+/**
+ *
+ * @param {import("express").Request} req
+ * @param {import("express").Response} res
+ */
+const findOrCreateChannelBySignedSender = async (req, res) => {
+  const client = StreamChat.getInstance(process.env.API_KEY, process.env.SECRET);
+  const {members} = req.body;
+
+  try {
+    if (!members.includes(req.userId)) members.push(req.userId);
+
+    if (members.length < 2) return ErrorResponse.e400(res, 'Members must be more than 1');
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This check for the length of `members` should be done before modifying the `members` array. If there's only one member and it's not `req.userId`, the current logic will add `req.userId` and bypass this check.
```diff
-    if (members.length < 2) return ErrorResponse.e400(res, 'Members must be more than 1');
+    if (allMembers.length < 2) return ErrorResponse.e400(res, 'Members must be more than 1');
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->